### ISSUE TITLE
Absolute beginner's question: how to remap multiple characters to multiple characters case-sensitively

### ISSUE THREAD

**Issue by J-Metsis** [tekezo/Karabiner#810]
I know there is a text substitution setting in System Preferences that allows me to have a text substitution ,say, from "rsvp" to "répondez s'il vous plaît". However, I have found that this doesn't work on all applications (e.g. Thunderbird). It seems to me that an application must be "aware of OS text substitution" (in lack of better wording). 

Enter Karabiner and editing private.xml… but this doesn't work for me:
<?xml version="1.0"?>
<root>
	<item>
		<identifier>private.rsvp</identifier>
		<name>RSVP</name>
		<autogen>__KeyToKey__ KeyCode::rsvp, KeyCode::répondez s'il vous plaît</autogen>
	</item>
</root>

More over, how to make this case-sensitively, i.e.
"bb" is remapped to "babe"
"BB" is remapped to "Big Brother"

Any pointers to documentation appreciated.

**Comment by dunkarooftop** [comment]
Karabiner cannot do what you wait, there are many text snippet for Mac out there.

aText
https://www.trankynam.com/atext/
is a good one, cheap too not like "TextExpender" from smile

**Comment by dunkarooftop** [comment]
Here is another one
http://www.ettoresoftware.com/mac-apps/typeit4me/

**Comment by J-Metsis** [comment]
Thank you for the info. I already knew a couple. My need is for a specific limited non-business purpose (one specific application that do not understand Mac's built-in text substitution; yes, I've left them a feature request), so for instance TypeIt4Me with a price tag of 20 € is overkill. Some are simply put not to my taste. Anyway, I take a spin with aText which seems to do what I want – and cheaply.

**Comment by dunkarooftop** [comment]
https://pqrs.org/osx/karabiner/xml.html.en

Here is the doc, you can see what Karabiner can do, scroll down to "autogen syntax"

You can do like
- Ctrl+"A" type ur snippet
- When double tap "A" type ur snippet
- When press "A" and "E" together type your snippet

to just name a few but with editing private.xml to edit your snippet is a lot of work, I say just pay the 5 bucks for aType. much better experience :)

Also if you happen to use Alfred already it has build-in snippet function too

**Comment by J-Metsis** [comment]
I've now tested a little aText and it seems to suit my needs, so I'll very likely buy that. Thank you for the recommendation. I appreciate tips where you point to another tool that suits better, instead of finding possibly clumsy circumvents or hacks in the tool originally referred.

The application where I need this and where Apple's built-in text substitution doesn't work, is based on Eclipse, which I also have. There is something in Eclipse and thereby in its derivate that prevents the built-in substitution to work. aText works in it.

I'm not afraid of editing XML (I write occasionally XML in my full-time job) but paying five USDs does beat writing XML if the result is less satisfactory.

Alfred was new to me (I had to google for it) and is overkill.
